### ISSUE TITLE
Specify how postfix:<.+>, postfix:<.*> works

### ISSUE THREAD

**Issue by W4anD0eR96** [rakudo/rakudo#1754]
Co. perl6/doc#1421

I've tried `say 42.+Str` and get `(42 Int<140709295767368>)`, which looks well.

But @Juerd states

> "$invocant.+method calls all methods called method" is inaccurate because it does not call all candidates of a multi method, even though they have the same name.

I fail to lookup its logic in src/. Thus come to here and ask for help.

Thank you.

**Comment by zoffixznet** [comment]
> I fail to lookup its logic in src/. Thus come to here and ask for help.

It's in https://github.com/rakudo/rakudo/blob/master/src/core/Mu.pm6#L846-L858

`$foo.+meth` walks the MRO and calls all the methods called `meth` and submethods called `meth` if the type is the same as type of `$foo`.

Those methods might be multies, in which case the matching candidate would be called.

> But @Juerd states [...] is inaccurate

Well, it is sorta-accurate, it does call all the methods, just not all candidates of all the methods.


```perl6
class A {
    method foo { say "from A"; }
}
class B is A {
    multi method foo { say "from B"; }
    multi method foo(Str) { say "from B (Str)"; }
}
class C is B is A {
    multi method foo { say "from C"; }
    multi method foo(Str) { say "from C (Str)"; }
}
```

```perl6
C.+foo;

# 2018.03.236 zoffix@VirtualBox~/R/rakudo (proto-fix-R1739)$ ./perl6 ../z.p6 
# from C
# from B
# from A
```

```perl6
C.+foo: "x";

# 2018.03.236 zoffix@VirtualBox~/R/rakudo (proto-fix-R1739)$ ./perl6 ../z.p6 
# from C (Str)
# from B (Str)
# Too many positionals passed; expected 1 argument but got 2
#   in method foo at ../z.p6 line 2
#   in block <unit> at ../z.p6 line 13
```

-------

Basically in pseudo-code it's:

```perl6
my @results;
given $invocant {
   @results.push: .has-meth-the-method ?? .meth-the-method !! .meth-the-submethod
}
for $invocant.^mro.skip {
    .has-meth-the-method and @result.push: $_.meth-the-method(with $invocant as self)
}
return @results;
```

**Comment by W4anD0eR96** [comment]
Thank you @zoffixznet .

So are these stuff internal so that we should not document them or they are public APIs happen to be rakudo specific so that we end up with documenting they somewhere.

**Comment by zoffixznet** [comment]
> So are these stuff internal so that we should not document them or they are public APIs happen to be rakudo specific so that we end up with documenting they somewhere.

The `.+` are public, but `.dispatch` is internal. Some of the `.dispatch` calls even get statically rewritten to more efficient code sometimes.
